pkg/controllers/kepler_internal.go

```diff
@@ -490,16 +538,79 @@ func exporterReconcilers(ki *v1alpha1.KeplerInternal, cluster k8s.Cluster) []rec
 	// namespace scoped
 	rs = append(rs, resourceReconcilers(updateResource,
 		exporter.NewServiceAccount(ki),
-		exporter.NewConfigMap(components.Full, ki),
-		exporter.NewDaemonSet(components.Full, ki),
 		exporter.NewService(ki),
 		exporter.NewServiceMonitor(ki),
 		exporter.NewPrometheusRule(ki),
 	)...)
+	if ki.Spec.Exporter.Redfish == nil {
+		rs = append(rs, resourceReconcilers(updateResource,
+			exporter.NewDaemonSet(components.Full, ki),
+			exporter.NewConfigMap(components.Full, ki),
+		)...)
+	} else {
+		rs = append(rs,
+			KeplerDaemonSetReconciler{
+				ki:      *ki,
+				updater: updateResource,
+				ds:      exporter.NewDaemonSet(components.Full, ki),
+			},
+			KeplerConfigMapReconciler{
+				ki:      *ki,
+				updater: updateResource,
+				cfm:     exporter.NewConfigMap(components.Full, ki),
+			},
+		)
+	}
 	rs = append(rs, resourceReconcilers(updateResource, openshiftNamespacedResources(ki, cluster)...)...)
 	return rs
 }
 
+type KeplerDaemonSetReconciler struct {
+	ki      v1alpha1.KeplerInternal
+	updater reconcileFn
+	ds      *appsv1.DaemonSet
+}
+
+func (r KeplerDaemonSetReconciler) Reconcile(ctx context.Context, cli client.Client, s *runtime.Scheme) reconciler.Result {
+
+	secretRef := r.ki.Spec.Exporter.Redfish.SecretRef
+	secret, err := r.getRedfishSecret(ctx, cli, secretRef)
```

Comment:
I think we should fail reconcile, set the status to available false,  if a secret is specified and is missing or invalid (missing the redfish.csv) . Otherwise there is no way for user to know if kepler is using redfish or not.